### PR TITLE
runtime: asymptote

### PR DESCRIPTION
**Pull request by avidseeker**
This adds the support for [Asymptote](https://asymptote.sourceforge.io/) Vector Graphics Language, and avoids packaging issues such as https://github.com/vectorgraphics/asymptote/issues/180.

Imported from [base/asy.vim](https://github.com/vectorgraphics/asymptote/blob/master/base/asy.vim).

### CODE REVIEW

File: runtime/ftplugin/asy.vim
Comment (dkearns) [inline runtime/ftplugin/asy.vim]: `makeprg` and `errorformat` should be moved to a dedicated compiler plugin.

File: runtime/syntax/asy.vim
Comment (zeertzjq) [inline runtime/syntax/asy.vim]: Wrong value

File: runtime/syntax/asy.vim
Comment (dkearns) [inline runtime/syntax/asy.vim]: Is Andy going to handle future updates to this repo?  If not could you add yourself as a co-maintainer, or even the sole maintainer if that's more appropriate?

File: runtime/syntax/asy.vim
Comment (dkearns) [inline runtime/syntax/asy.vim:12]: This block can be reduced to the following.  Vim 6 was released over twenty years ago and the special handling for Vim 5 was removed from the distributed files some time ago.

```vim
if exists("b:current_syntax")
  finish
endif
```

File: runtime/syntax/asy.vim
Comment (dkearns) [inline runtime/syntax/asy.vim]:
```suggestion
let b:current_syntax = "asy"
```

File: runtime/syntax/asy.vim
Comment (dkearns) [inline runtime/syntax/asy.vim]: This conditional can be removed along with the use of the `HiLink` command and replaced with `hi def link`.   See other syntax files for an example.

File: runtime/syntax/asy.vim
Comment (dkearns) [inline runtime/syntax/asy.vim:35]: It would be nice if these config variables were documented at `:help ft-asy-syntax`.

**Review by dkearns — CHANGES_REQUESTED** [review]
Thanks.

Could you also add entries to https://github.com/vim/vim/blob/master/.github/MAINTAINERS?

**Comment by chrisbra** [comment]
Thanks. Can you please also add a test to testdir/test_filetype.vim?

File: runtime/syntax/asy.vim
Comment (clason) [reply runtime/syntax/asy.vim:260]: Why? Asymptote is basically C.

File: runtime/syntax/asy.vim
Comment (avidseeker) [reply runtime/syntax/asy.vim]: Last change to that file is [3 years ago](https://github.com/vectorgraphics/asymptote/commits/master/base/asy.vim). I'll add myself as a maintainer.

File: runtime/filetype.vim
Comment (zeertzjq) [inline runtime/filetype.vim:115]: This should be `asymptote` now

File: runtime/syntax/asymptote.vim
Comment (zeertzjq) [inline runtime/syntax/asymptote.vim]: ditto

File: src/testdir/test_filetype.vim
Comment (zeertzjq) [inline src/testdir/test_filetype.vim:110]: ditto

File: runtime/filetype.vim
Comment (avidseeker) [reply runtime/filetype.vim:115]: You mean `file.asymptote`? The common extension is `.asy`. I changed *the name* of the filetype which is different from the extension. E.g: filetype name of `.md` is markdown, and no one use `file.markdown` as a filename.

Comment (zeertzjq) [reply runtime/filetype.vim:115]: `setf asymptote`

File: src/testdir/test_filetype.vim
Comment (avidseeker) [reply src/testdir/test_filetype.vim:110]:
```
    asy: ['file.asymptote', 'file.asy'],
```
Both should be included, no?

Comment (zeertzjq) [reply src/testdir/test_filetype.vim:110]: No,
```
    asymptote: ['file.asy'],
```

File: runtime/doc/syntax.txt
Comment (dkearns) [inline runtime/doc/syntax.txt]:
```suggestion
By default, Asymptote-defined colors (e.g: lightblue) are highlighted. To
```

**Review by dkearns — CHANGES_REQUESTED** [review]
Thanks for the changes.

Is the filetype name change going to cause problems for existing users of the upstream repo?  Will it be changed there also?

**Comment by avidseeker** [comment]
I have in mind submitting a PR to asymptote and packagers (Fedora) with the changes after merging this. Asymptote suggests to user copying `asy.vim` to their user config, so this might be a breaking change.

**Comment by chrisbra** [comment]
> If it's already out there in the wild I think it would be better not to change the name.

Yeah, agree. Let't not change it please.

File: runtime/filetype.vim
Comment (dkearns) [inline runtime/filetype.vim]:
```suggestion
au BufNewFile,BufRead *.asy		setf asy
```

File: runtime/doc/syntax.txt
Comment (dkearns) [inline runtime/doc/syntax.txt]:
```suggestion
ASYMPTOTE					*asy.vim* *ft-asy-syntax*
```

I think it probably makes most sense to title the section with the proper name and use the actual filetype in the tags.  The PPWizard (ppwiz) section does this.

The filename is also usually included in the tag list.  I know it's not in the docs directly above this entry.

**Review by dkearns — CHANGES_REQUESTED** [review]

**Review by dkearns — APPROVED** [review]
Looks good, thanks.

Sorry for the naming merry-go-round.

**Pull request closed without merging**
